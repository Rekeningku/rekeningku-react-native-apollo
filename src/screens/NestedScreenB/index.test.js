import React from 'react'
import sinon from 'sinon'
import { shallow } from 'enzyme'
import { NestedScreenB } from './index'


describe('NestedScreenB Component', () => {
  it('renders with alert button', () => {
    const fakeHandler = sinon.fake()
    const wrapper = shallow(
      <NestedScreenB
        shouldShowAlertButton
        handleHideAlertButton={fakeHandler}
      />,
    )
    wrapper.find('ButtonComponent').at(1).simulate('press')

    expect(fakeHandler.callCount).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders without alert button', () => {
    const fakeHandler = sinon.fake()
    const wrapper = shallow(
      <NestedScreenB
        shouldShowAlertButton={false}
        handleShowAlertButton={fakeHandler}
      />,
    )
    wrapper.find('ButtonComponent').at(1).simulate('press')

    expect(fakeHandler.callCount).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('calls pop screen handler', () => {
    const fakeHandler = sinon.fake()
    const wrapper = shallow(<NestedScreenB handlePopToRoot={fakeHandler} />)
    wrapper.find('ButtonComponent').at(0).simulate('press')

    expect(fakeHandler.callCount).toEqual(1)
  })
})
