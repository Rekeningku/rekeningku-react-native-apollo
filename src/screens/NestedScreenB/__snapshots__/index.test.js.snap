// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedScreenB Component renders with alert button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NestedScreenB
    handleHideAlertButton={[Function]}
    shouldShowAlertButton={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Component)>
          Screen B
        </Styled(Component)>,
        <Styled(Component)>
          This is the end of the road. You can press the top left button to get to the previous screen.
        </Styled(Component)>,
        <Styled(Component)>
           
          On IOS you can also swipe from left to right in order close this screen.
        </Styled(Component)>,
        <Styled(Component)>
          If you want to go directly to the home screen then press the button bellow.
        </Styled(Component)>,
        <ButtonComponent>
          Pop
        </ButtonComponent>,
        <Styled(Component)>
          You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.
        </Styled(Component)>,
        <ButtonComponent
          onPress={[Function]}
        >
          HIDE Top Right Button
        </ButtonComponent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Screen B",
        },
        "ref": null,
        "rendered": "Screen B",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "This is the end of the road. You can press the top left button to get to the previous screen.",
        },
        "ref": null,
        "rendered": "This is the end of the road. You can press the top left button to get to the previous screen.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            "On IOS you can also swipe from left to right in order close this screen.",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "On IOS you can also swipe from left to right in order close this screen.",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "If you want to go directly to the home screen then press the button bellow.",
        },
        "ref": null,
        "rendered": "If you want to go directly to the home screen then press the button bellow.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Pop",
          "onPress": undefined,
        },
        "ref": null,
        "rendered": "Pop",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.",
        },
        "ref": null,
        "rendered": "You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "HIDE Top Right Button",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": "HIDE Top Right Button",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Component)>
            Screen B
          </Styled(Component)>,
          <Styled(Component)>
            This is the end of the road. You can press the top left button to get to the previous screen.
          </Styled(Component)>,
          <Styled(Component)>
             
            On IOS you can also swipe from left to right in order close this screen.
          </Styled(Component)>,
          <Styled(Component)>
            If you want to go directly to the home screen then press the button bellow.
          </Styled(Component)>,
          <ButtonComponent>
            Pop
          </ButtonComponent>,
          <Styled(Component)>
            You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.
          </Styled(Component)>,
          <ButtonComponent
            onPress={[Function]}
          >
            HIDE Top Right Button
          </ButtonComponent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Screen B",
          },
          "ref": null,
          "rendered": "Screen B",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "This is the end of the road. You can press the top left button to get to the previous screen.",
          },
          "ref": null,
          "rendered": "This is the end of the road. You can press the top left button to get to the previous screen.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              "On IOS you can also swipe from left to right in order close this screen.",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "On IOS you can also swipe from left to right in order close this screen.",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "If you want to go directly to the home screen then press the button bellow.",
          },
          "ref": null,
          "rendered": "If you want to go directly to the home screen then press the button bellow.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Pop",
            "onPress": undefined,
          },
          "ref": null,
          "rendered": "Pop",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.",
          },
          "ref": null,
          "rendered": "You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "HIDE Top Right Button",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "HIDE Top Right Button",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`NestedScreenB Component renders without alert button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NestedScreenB
    handleShowAlertButton={[Function]}
    shouldShowAlertButton={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Component)>
          Screen B
        </Styled(Component)>,
        <Styled(Component)>
          This is the end of the road. You can press the top left button to get to the previous screen.
        </Styled(Component)>,
        <Styled(Component)>
           
          On IOS you can also swipe from left to right in order close this screen.
        </Styled(Component)>,
        <Styled(Component)>
          If you want to go directly to the home screen then press the button bellow.
        </Styled(Component)>,
        <ButtonComponent>
          Pop
        </ButtonComponent>,
        <Styled(Component)>
          You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.
        </Styled(Component)>,
        <ButtonComponent
          onPress={[Function]}
        >
          SHOW Top Right Button
        </ButtonComponent>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Screen B",
        },
        "ref": null,
        "rendered": "Screen B",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "This is the end of the road. You can press the top left button to get to the previous screen.",
        },
        "ref": null,
        "rendered": "This is the end of the road. You can press the top left button to get to the previous screen.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            " ",
            "On IOS you can also swipe from left to right in order close this screen.",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "On IOS you can also swipe from left to right in order close this screen.",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "If you want to go directly to the home screen then press the button bellow.",
        },
        "ref": null,
        "rendered": "If you want to go directly to the home screen then press the button bellow.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Pop",
          "onPress": undefined,
        },
        "ref": null,
        "rendered": "Pop",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.",
        },
        "ref": null,
        "rendered": "You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "SHOW Top Right Button",
          "onPress": [Function],
        },
        "ref": null,
        "rendered": "SHOW Top Right Button",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Component)>
            Screen B
          </Styled(Component)>,
          <Styled(Component)>
            This is the end of the road. You can press the top left button to get to the previous screen.
          </Styled(Component)>,
          <Styled(Component)>
             
            On IOS you can also swipe from left to right in order close this screen.
          </Styled(Component)>,
          <Styled(Component)>
            If you want to go directly to the home screen then press the button bellow.
          </Styled(Component)>,
          <ButtonComponent>
            Pop
          </ButtonComponent>,
          <Styled(Component)>
            You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.
          </Styled(Component)>,
          <ButtonComponent
            onPress={[Function]}
          >
            SHOW Top Right Button
          </ButtonComponent>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Screen B",
          },
          "ref": null,
          "rendered": "Screen B",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "This is the end of the road. You can press the top left button to get to the previous screen.",
          },
          "ref": null,
          "rendered": "This is the end of the road. You can press the top left button to get to the previous screen.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              " ",
              "On IOS you can also swipe from left to right in order close this screen.",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "On IOS you can also swipe from left to right in order close this screen.",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "If you want to go directly to the home screen then press the button bellow.",
          },
          "ref": null,
          "rendered": "If you want to go directly to the home screen then press the button bellow.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Pop",
            "onPress": undefined,
          },
          "ref": null,
          "rendered": "Pop",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.",
          },
          "ref": null,
          "rendered": "You can do some cool magic with Wix navigation. You can for example dynamically set navigation buttons.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "SHOW Top Right Button",
            "onPress": [Function],
          },
          "ref": null,
          "rendered": "SHOW Top Right Button",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
